### ISSUE TITLE
BUG: v2.46.0 introduced a postinstall lifecycle script which requires devdependencies for non-development installations

### ISSUE THREAD

**Issue by michael-kirkegaard-solita** [apigee/apigeelint#407]
In version v2.46.0 the package.json file has introduced the lifecycle script `postinstall`.

The `postinstall` script executes cli commands that requires the packages "peggy" and "mkdirp" to be installed.
The packages "peggy" and "mkdirp" are not defined in the dependencies, only in the devDependencies.

When apigeelint in installed for non-development usage with `npm install -g apigeelint@latest`, devDependencies are not installed [1] but the `postinstall` script is triggered [2], resulting in the `postinstall` script failing because "peggy" and "mkdirp" are not necessarily installed.

This makes the `npm install` command fail.

An example of this:
![image](https://github.com/apigee/apigeelint/assets/119622241/e324873c-09d3-4471-b336-2f4cc9f6c204)

References:
[1]: https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file
[2]: https://docs.npmjs.com/cli/v8/using-npm/scripts#npm-install

**Comment by gislikonrad** [comment]
Yeah, I noticed this too, but I think the actual problem is that the postinstall script is running from the ./node_modules/apigeelint folder and from there it's referencing ./node_modules/peggy/peggy.js. I tried manually adding peggy to my project, but it still failed because the ./node_modules/apigeelint folder didn't have it's own node_modules folder. 

Adding an `npm i` to the post install script should solve this, but I don't know how clean that would be.
